static_frame/core/index_hierarchy.py

```diff
@@ -237,31 +243,31 @@ class IndexHierarchy(IndexBase):
             '_pending_extensions',
             )
 
-    _indices: tp.List[Index] # Of index objects
-    _indexers: NDArrayAny # 2D - integer arrays
-    _name: NameType
+    _indices: tp.List[Index[tp.Any]] # Of index objects
+    _indexers: TNDArrayAny # 2D - integer arrays
+    _name: TName
     _blocks: TypeBlocks
     _recache: bool
-    _values: tp.Optional[NDArrayAny] # Used to cache the property `values`
+    _values: tp.Optional[TNDArrayAny] # Used to cache the property `values`
     _map: HierarchicalLocMap
-    _index_types: tp.Optional['Series'] # Used to cache the property `index_types`
-    _pending_extensions: tp.Optional[tp.List[tp.Union[SingleLabelType, 'IndexHierarchy', PendingRow]]]
+    _index_types: tp.Optional[Series[tp.Any, np.object_]] # Used to cache the property `index_types`
+    _pending_extensions: tp.Optional[tp.List[tp.Union[TSingleLabel, 'IndexHierarchy', PendingRow]]]
 
     # _IMMUTABLE_CONSTRUCTOR is None from IndexBase
     # _MUTABLE_CONSTRUCTOR will be defined after IndexHierarhcyGO defined
 
-    _INDEX_CONSTRUCTOR: IndexConstructor = Index
+    _INDEX_CONSTRUCTOR: tp.Callable[..., 'IndexBase'] = Index
```

Comment:
How come it's not `TIndexCtor`?

Reply:
Yeah,  that seems odd. That type should be as narrow as possible, as it does not need to support the general case. So the right thing to do is just remove the hint (as done in similar args elsewhere) and let it be the type of the argument.